vantage6/cli/rabbitmq/queue_manager.py

```diff
@@ -0,0 +1,222 @@
+    )
+
+
+class RabbitMQManager:
+    """
+    Manages the RabbitMQ docker container
+    """
+    def __init__(self, ctx: ServerContext, image: str = None) -> None:
+        """
+        Parameters
+        ----------
+        ctx: ServerContext
+            Configuration object
+        queue_uri: str
+            URI where the RabbitMQ instance should be running
+        """
+        self.ctx = ctx
+        rabbit_settings = self.ctx.config.get('rabbitmq')
+        self.rabbit_user = rabbit_settings['user']
+        self.rabbit_pass = rabbit_settings['password']
+
+        self.queue_uri = get_rabbitmq_uri(rabbit_settings)
+        self.docker = docker.from_env()
+        self.image = image if image else DEFAULT_RABBIT_IMAGE
+        self.rabbit_container_name = f'{APPNAME}-{ctx.name}-rabbitmq'
+
+    def start(self) -> None:
+        """
+        Start a docker container which runs a RabbitMQ queue
+        """
+        # get volumes which contain rabbitMQ configuration and a location to
+        # save rabbitMQ files so that they persist after container is stopped
+        volumes = self._get_volumes()
+
+        # expose port 5672 inside the container as port 5672 on the host, and
+        # same for 15672 in container to 8080 on host
+        # TODO check if these ports are not already used on the host
+        ports = {
+            f'{QUEUE_PORT}/tcp': QUEUE_PORT,
+            # TODO this is for the management tool, do we keep this? Not used
+            # at the moment..
+            '15672/tcp': 8080
+        }
+
+        # if a RabbitMQ container is already running, kill and remove it
+        remove_container_if_exists(
+            docker_client=self.docker, name=self.rabbit_container_name
+        )
+
+        # run rabbitMQ container
+        self.rabbit_container = self.docker.containers.run(
+            name=self.rabbit_container_name,
+            image=self.image,
+            volumes=volumes,
+            ports=ports,
+            detach=True,
+            restart_policy={"Name": "always"},
+            hostname=f'{APPNAME}-{self.ctx.name}-rabbit',
+            labels={
+                f"{APPNAME}-type": "rabbitmq",
+            }
+        )
+
+        # Wait until RabbitMQ is up before continuing with other stuff
+        self._wait_for_startup()
+
+    def _wait_for_startup(self) -> None:
+        """ Wait until RabbitMQ has been initialized """
+        # TODO make the time / # attempts settable? Is this enough time (5m)?
+        ATTEMPTS = 31
+        is_running = False
+        for _ in range(ATTEMPTS):
+            if self.is_running():
+                is_running = True
+                break
+            debug("RabbitMQ is not yet running, trying again in 10s...")
+            time.sleep(10)
+        if is_running:
+            info("RabbitMQ was started successfully!")
+        else:
+            # TODO exit or disable?
+            error("Could not start RabbitMQ! Exiting...")
+            exit(0)
```

Comment:
i think an `exit(1)` would be more appriopiate as `0` is generally considered a succesfull termination as if it is intended